apps/src/templates/VerticalImageResourceCard.jsx

```diff
@@ -155,6 +170,7 @@ class VerticalImageResourceCard extends Component {
             color={Button.ButtonColor.gray}
             text={buttonText}
             style={[styles.button, localeStyle]}
+            onClick={this.logToFirehose}
```

Comment:
Sorry for not thinking of this earlier (and credit to @joshlory for thinking of it during triage) but there are reporting data-loss risks with this approach.  While the click might start a network operation to report, the navigation to a new page in the same tab is likely to actually cancel the network operation before it completes.  

Long ago we once manually built a kind of solution to this where we'd do the reporting first (as a Google Analytics event in that case) and then did the navigation in JS in the callback, but this has a risk of taking a long time to do the navigate.  We don't currently have an in-house generalised solution to this.

I've noticed that Google Tag manager does have support for this situation by offering to wait to do the navigate on a regular link click until its own reporting has finished or until a specific timeout has occurred, whichever happens first:

![Screenshot 2019-12-04 12 51 36](https://user-images.githubusercontent.com/2205926/70180690-5f1e9780-1695-11ea-9c0a-bc61ad350caa.png)

Reply:
> Sorry for not thinking of this earlier (and credit to @joshlory for thinking of it during triage) but there are reporting data-loss risks with this approach. While the click might start a network operation to report, the navigation to a new page in the same tab is likely to actually cancel the network operation before it completes.
> 
> Long ago we once manually built a kind of solution to this where we'd do the reporting first (as a Google Analytics event in that case) and then did the navigation in JS in the callback, but this has a risk of taking a long time to do the navigate. We don't currently have an in-house generalised solution to this.
> 
> I've noticed that Google Tag manager does have support for this situation by offering to wait to do the navigate on a regular link click until its own reporting has finished or until a specific timeout has occurred, whichever happens first:
> 
> ![Screenshot 2019-12-04 12 51 36](https://user-images.githubusercontent.com/2205926/70180690-5f1e9780-1695-11ea-9c0a-bc61ad350caa.png)



> the navigation to a new page in the same tab is likely to actually cancel the network operation before it completes.

@breville If I have a button or link that I navigates to a _new_ tab can I log to firehose without risking data-loss?

Reply:
Yes, I believe the new page will open while the current page will do its reporting.

Reply:
@clareconstantine more context about the Firehose v. Google Tag Manager issue

Reply:
thanks!